### DISCUSSION TITLE
[Atom] Attribute in generic classes

### DISCUSSION THREAD

**Discussion by Girildo** [codewriter-packages/UniMob#26] [Q&A] [answered]
I was wondering what is the reason behind the fact that it's not possible to use [Atom] in generic classes.
I am using UniMob.UI and it'd be cool to create type parametrized widgets.

Is there an unsoromountable technical limitation, is it a matter of interest/implementation time, anything else?

Could one possibly work around the limitation, currently?

Thank you once more for the awesome library!

**Comment by vanifatovvlad** [comment, marked as answer]
UniMob library under the hood[ generates IL code](https://github.com/codewriter-packages/UniMob/blob/master/CodeGen/AtomWeaverV2.cs) using the Mono.Cecil library. This is quite simple for regular classes, but the implementation for generic classes is much more complicated. I tried to add support for it, but I couldn't make a correct code generator. In my projects, I did not need generics, so this feature was simply postponed.
